src/memray/commands/run.py

```diff
@@ -66,24 +66,17 @@ def _run_tracker(
                 print(post_run_message)
 
 
-def _child_process(
-    port: int,
-    native: bool,
-    run_as_module: bool,
-    run_as_cmd: bool,
-    quiet: bool,
-    script: str,
-    script_args: List[str],
-) -> None:
+def _child_process() -> None:
+    params = json.loads(sys.stdin.read())
     args = argparse.Namespace(
-        native=native,
-        run_as_module=run_as_module,
-        run_as_cmd=run_as_cmd,
-        quiet=quiet,
-        script=script,
-        script_args=script_args,
+        native=params["native"],
```

Comment:
Ugh, this is a bit ugly, tbh. I would prefer to only scape the stuff that we need to scape instead of affecting all parameters. I don't feel very strongly, though, in case you do

Reply:
I'm not sure I follow what you mean. If you think that the previous approach - formatting user-provided strings into a string of Python source code - was cleaner than the approach of serializing parameters and sending them to the process over stdin, I'm shocked.

I also don't want some hybrid of the two approaches where we handle fields differently depending on whether or not they contain strings.

But if you do prefer the previous approach, it should be possible to make it work correctly, instead - the Python injection approach isn't intrinsically wrong, it just requires correct escaping...

Reply:
> I'm not sure I follow what you mean. If you think that the previous approach - formatting user-provided strings into a string of Python source code - was cleaner than the approach of serializing parameters and sending them to the process over stdin, I'm shocked.

No, I am saying that we are jsonifiying all arguments here where a bunch of them are just `bools`. If in the future we need to pass some Python object that cannot be serialized this may be annoying so I am advocating to only serialize the arguments that need to be sanitized.

Reply:
Yeah, I find it a bit cleaner than adding tons of dict access, but I am happy to go with the other if you prefer it

Reply:
> I don't like that this leaves `ps` looking like:

If you are referring to the `_child_process(49241,False,False,True,False`, that's a good point, but one that's fixable with keyword arguments :)

Reply:
I'm more talking about the fact that it embeds the entire state of an `argparse.parse` call into command line arguments that show up in `ps`. It feels like it's leaking implementation details. I also don't like that it takes up the entire width of my screen, it's weird.

But that's mostly personal preference, I guess, and I'm happy to call it out of scope for the bug fix.

Reply:
> Do you prefer https://github.com/bloomberg/memray/commit/c923a7541f1b7b5b8148dd29759604cffa06c5ae ?

Make that https://github.com/bloomberg/memray/pull/147/commits/bacbb94ae4da66f660dfff5ef338f0d5547dc1db (amended because I missed the news entry for the bugfix).